homeassistant/components/homekit/type_switches.py

```diff
@@ -218,10 +224,13 @@ class Valve(HomeAccessory):
     def __init__(self, *args: Any) -> None:
         """Initialize a Valve accessory object."""
         super().__init__(*args)
+        self.domain = split_entity_id(self.entity_id)[0]
         state = self.hass.states.get(self.entity_id)
         assert state
 
-        valve_type = self.config[CONF_TYPE]
+        valve_type = (
+            TYPE_VALVE if self.domain == VALVE_DOMAIN else self.config[CONF_TYPE]
+        )
```

Comment:
With all the branching here, it probably makes sense to subclass `Valve` so we can do all the branches in each subclass

Reply:
That was also my initial thought but went eventually for this implementation as the switch with type valve is already there and should continue to work. In opinion this is the cleanest solution without requiring larger code changes.

Of course we can create a base class type for shared functionality and subclass switch and valve, but I think it's bit overkill here.

Reply:
I would prefer not to have to maintain the branching long term especially since valve and switch are completely separate platforms it's likely they may diverge more in the future.  Please split the classes with a base class.

Reply:
Ok, I understand. I expect to have time to do this within 1 or 2 weeks.

Reply:
@bdraco - Please check the latest commit.

Reply:
I adjusted the classes to reduce duplicate code.

Please see the comment I added about the test.

Thanks

Reply:
I have split the tests. Merged with your last commit. And did some quick testing from within HA/HomeKit.